packages/@react-stately/grid/src/useGridState.ts

```diff
@@ -39,6 +40,15 @@ export function useGridState<T extends object, C extends GridCollection<T>>(prop
           key = children[0]?.key;
         }
       }
+    } else if (preventCellFocus && key != null) {
+      // If preventCellFocus is true, set focus to the row containing the cell
+      let item = collection.getItem(key);
+      if (item?.type === 'cell') {
+        while (item && item?.type === 'cell') {
+          item = collection.getItem(item.parentKey);
+        }
+        key = item.key;
+      }
```

Comment:
Bit iffy about needing to add a new prop to force focus to the row if the grid cell is focused. Had to do this to handle the case where the user clicks on the CheckBox directly which causes `useGridCell` + `useSelectableItem` to try to set the focused key as the cell itself instead of the row. Open to feedback on if this is the best place to put this.

Gonna try keeping the focus on the cell in a separate PR to see if there is a nicer way to circumvent Checkbox focusing.

Reply:
Approach 2 here: https://github.com/adobe/react-spectrum/pull/2737